pwiz_tools/Skyline/Model/SkypFile.cs

```diff
@@ -19,67 +20,108 @@ public class SkypFile
-                            SrmDocumentSharing.EXT_SKY_ZIP, urlString));
+                        // First line in the skyp file is the URL of the skyp.zip file on a Panorama server
+                        skyp.SkylineDocUri = GetSkyFileUrl(line);
+                        first = false;
                     }
-
-                    var validUrl = Uri.TryCreate(urlString, UriKind.Absolute, out var skyFileUri)
-                                   && (skyFileUri.Scheme == Uri.UriSchemeHttp || skyFileUri.Scheme == Uri.UriSchemeHttps);
-                    if (!validUrl)
+                    else
                     {
-                        throw new InvalidDataException(string.Format(
-                            Resources.SkypFile_GetSkyFileUrl__0__is_not_a_valid_URL_on_a_Panorama_server_, urlString));
+                        // The remaining lines should contain colon separated key value pairs, e.g. FileSize:475831
+                        var parts = line.Split(':');
+                        if (parts.Length == 2)
+                        {
+                            var key = parts[0].Trim();
+                            var value = parts[1].Trim();
+                            if (@"FileSize".Equals(key) && value.Length > 0)
+                            {
+                                if (long.TryParse(value, out var size) && size > 0)
+                                {
+                                    skyp.Size = size;
+                                }
+                            }
+                            else if (@"DownloadingUser".Equals(key) && value.Length > 0)
+                            {
+                                skyp.DownloadingUser = value;
+                            }
+                        }
                     }
-
-                    return skyFileUri;
                 }
             }
+            if (skyp.SkylineDocUri == null)
+            {
+                throw new InvalidDataException(string.Format(
+                    Resources
+                        .SkypFile_GetSkyFileUrl_File_does_not_contain_the_URL_of_a_shared_Skyline_archive_file___0___on_a_Panorama_server_,
+                    SrmDocumentSharing.EXT_SKY_ZIP));
+            }
+        }
 
-            throw new InvalidDataException(string.Format(
-                Resources.SkypFile_GetSkyFileUrl_File_does_not_contain_the_URL_of_a_shared_Skyline_archive_file___0___on_a_Panorama_server_,
-                SrmDocumentSharing.EXT_SKY_ZIP));
+        private static Uri GetSkyFileUrl(string line)
+        {
+            var urlString = line;
+            if (!urlString.EndsWith(SrmDocumentSharing.EXT))
+            {
+                // This is not a shared Skyline zip.
+                throw new InvalidDataException(string.Format(
+                    Resources
+                        .SkypFile_GetSkyFileUrl_Expected_the_URL_of_a_shared_Skyline_document_archive___0___in_the_skyp_file__Found__1__instead_,
```

Comment:
Maybe again separate the file name from the full URL. Most of the URL is extra information. And "in the skyp file" is probably also not necessary.